File: appinventor/components/src/com/google/appinventor/components/runtime/ChartComponent.java
// -*- mode: java; c-basic-offset: 2; -*-
// Copyright 2023-2024 MIT, All rights reserved
// Released under the Apache License, Version 2.0
// http://www.apache.org/licenses/LICENSE-2.0

package com.google.appinventor.components.runtime;

/**
 * The ChartComponent interface provides methods common to all chart components. Any component
 * intended to be added to a {@link Chart} MUST implement this interface.
 */
public interface ChartComponent extends Component {
  /**
   * Initialize the component's chart data object.
   */
  void initChartData();
}

File: appinventor/components-ios/src/ChartComponent.swift
// -*- mode: swift; swift-mode:basic-offset: 2; -*-
// Copyright © 2022 Massachusetts Institute of Technology, All rights reserved.
// Released under the Apache License, Version 2.0
// http://www.apache.org/licenses/LICENSE-2.0

import Foundation

// Define a protocol to represent the common functionalities for chart components
protocol ChartComponent: Component {
  // Method to initialize the chart data object
  func initChartData()
}
